pkg/nvcdi/lib-nvml.go

```diff
@@ -72,14 +72,18 @@ func (l *nvmllib) getDeviceSpecGeneratorsForIDs(ids ...string) (DeviceSpecGenera
 		identifiers = append(identifiers, device.Identifier(id))
 	}
 
-	devices, err := l.getNVMLDevicesByID(identifiers...)
+	uuids, err := l.normalizeDeviceIDs(identifiers...)
 	if err != nil {
 		return nil, err
 	}
 
 	var DeviceSpecGenerators DeviceSpecGenerators
-	for i, device := range devices {
-		editor, err := l.newDeviceSpecGeneratorFromNVMLDevice(ids[i], device)
+	for _, uuid := range uuids {
+		device, ret := l.nvmllib.DeviceGetHandleByUUID(string(uuid))
+		if ret != nvml.SUCCESS {
+			return nil, fmt.Errorf("failed to get device handle from UUID: %v", ret)
+		}
+		editor, err := l.newDeviceSpecGeneratorFromNVMLDevice(string(uuid), device)
```

Comment:
not for this PR but, why we call the variable `editor` if the func name is `newGenerator`, var name should be generator.